src/main/scala/se/nimsa/sbx/filtering/FilteringServiceActor.scala

```diff
@@ -0,0 +1,82 @@
+package se.nimsa.sbx.filtering
+
+import akka.actor.{Actor, Props}
+import akka.event.{Logging, LoggingReceive}
+import akka.util.Timeout
+import se.nimsa.sbx.app.GeneralProtocol.{Source, SourceDeleted, SourceRef}
+import se.nimsa.sbx.filtering.FilteringProtocol._
+import se.nimsa.sbx.util.FutureUtil.await
+
+import scala.concurrent.Future
+
+class FilteringServiceActor(filteringDAO: FilteringDAO)(implicit timeout: Timeout) extends Actor {
+  val log = Logging(context.system, this)
+
+  implicit val system = context.system
+  implicit val ec = context.dispatcher
+
+  val emptyTagPathsSeq: Seq[TagFilterTagPath] = Seq()
+
+  override def preStart(): Unit =
+    context.system.eventStream.subscribe(context.self, classOf[SourceDeleted])
+
+  def receive: Receive = LoggingReceive {
+    case AddTagFilter(tagFilterSpec) =>
+      sender ! insertTagFilter(tagFilterSpec)
+    case gtf: GetTagFilters =>
+      sender ! getTagFilters(gtf.startIndex, gtf.count)
+    case RemoveTagFilter(tagFilterId) =>
+      sender ! removeTagFilter(tagFilterId)
+    case GetTagFilter(tagFilterId) =>
+      sender ! getTagFilter(tagFilterId)
+    case GetFilterForSource(source) =>
+      sender ! getTagFilterForSource(source)
+    case SetFilterForSource(source, tagFilterId) =>
+      sender ! setTagFilterForSource(source, tagFilterId)
+//    case RemoveFilterForSource(sourceFilterId) =>
+//      sender ! removeFilterForSource(sourceFilterId)
+    case RemoveFilterForSource(sourceRef) =>
+      sender ! removeFilterForSource(sourceRef)
+    case SourceDeleted(sourceRef) =>
+      removeFilterForSource(sourceRef)
+  }
+
+  def insertTagFilter(tagFilterSpec: TagFilterSpec): TagFilterAdded = {
+    await(filteringDAO.createOrUpdateTagFilter(tagFilterSpec).map(TagFilterAdded(_)))
```

Comment:
It would be great if we avoid await in new code. We're trying to phase use of this out everywhere. Mix in the train `SequentialPipeToSupport` in this actor and use `.pipeTo(sender)` for calls that only read from the database and `.pipeSequentiallyTo(sender)` for any calls that update the database.